File: src/AutomatedBikeSystem.java
public class AutomatedBikeSystem {
    private boolean isOn = false;
    private int acceleration = 0;


    public void togglePower() {
        isOn = !isOn;
    }

    public boolean isBikeOn() {
        return isOn;
    }

    public int getAcceleration() {
        return acceleration;
    }

    public void increaseAcceleration() {
        if (!isOn) return;

        final int maxAcceleration = 70;

        if (acceleration >= 0 && acceleration <= 20) {
            acceleration += 1;
        } else if (acceleration >= 21 && acceleration <= 30) {
            acceleration += 2;
        } else if (acceleration >= 31 && acceleration <= 40) {
            acceleration += 3;
        } else if (acceleration >= 41 && acceleration < maxAcceleration) {
            acceleration += 4;
        }

        if (acceleration > maxAcceleration) {
            acceleration = maxAcceleration;
        }
    }

    public void decreaseAcceleration() {
        if (!isOn) return;

        final int minAcceleration = 0;

        if (acceleration >= 41 && acceleration <= 70) {
            acceleration -= 4;
        } else if (acceleration >= 31 && acceleration <= 40) {
            acceleration -= 3;
        } else if (acceleration >= 21 && acceleration <= 30) {
            acceleration -= 2;
        } else if (acceleration >= 1 && acceleration <= 20) {
            acceleration -= 1;
        }

        if (acceleration < minAcceleration) {
            acceleration = minAcceleration;
        }
    }

}

File: PYTHON/automated_bike_system.py
class AutomatedBikeSystem:
    MAX_ACCELERATION = 70
    MIN_ACCELERATION = 0

    def __init__(self, acceleration: int = 0, power_toggle: bool = False) -> None:
        self.acceleration = acceleration
        self.isOn = power_toggle

    def power_toggle(self) -> None:
        self.isOn = not self.isOn

    def is_on(self) -> bool:
        return self.isOn

    def get_acceleration(self) -> int:
        return self.acceleration

    def increase_acceleration(self) -> None:
        if not self.isOn:
            return

        if 0 <= self.acceleration <= 20:
            self.acceleration += 1
        elif 21 <= self.acceleration <= 30:
            self.acceleration += 2
        elif 31 <= self.acceleration <= 40:
            self.acceleration += 3
        elif 41 <= self.acceleration < self.MAX_ACCELERATION:
            self.acceleration += 4

        if self.acceleration > self.MAX_ACCELERATION:
            self.acceleration = self.MAX_ACCELERATION

    def decrease_acceleration(self) -> None:
        if not self.isOn:
            return

        if 41 <= self.acceleration <= 70:
            self.acceleration -= 4
        elif 31 <= self.acceleration <= 40:
            self.acceleration -= 3
        elif 21 <= self.acceleration <= 30:
            self.acceleration -= 2
        elif 1 <= self.acceleration <= 19:
            self.acceleration -= 1

        if self.acceleration < self.MIN_ACCELERATION:
            self.acceleration = self.MIN_ACCELERATION

if __name__ == '__main__':
    main = AutomatedBikeSystem()
